src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php

```diff
@@ -154,4 +177,33 @@ protected function generateRegionNames(ArrayAccessibleResourceBundle $localeBund
 
         return $regionNames;
     }
+
+    protected function generateAlpha3(ArrayAccessibleResourceBundle $metadataBundle)
+    {
+        $alpha2Codes = array_flip($this->regionCodes);
+        $alpha2ToAlpha3 = [];
+        foreach ($metadataBundle['alias']['territory'] as $alias => $data) {
+            if (3 === \strlen($alias) && 'overlong' === $data['reason'] && !ctype_digit($alias)) {
+                $alpha2Code = $data['replacement'];
+                if (isset($alpha2Codes[$alpha2Code])) {
+                    if (!isset($alpha2ToAlpha3[$alpha2Code])) {
+                        $alpha2ToAlpha3[$alpha2Code] = $alias;
+                    } else {
+                        // Found a second alias for the same country
+                        if (isset(self::$preferredAlpha2ToAlpha3Mapping[$alpha2Code])) {
+                            $preferred = self::$preferredAlpha2ToAlpha3Mapping[$alpha2Code];
+                            // Only use the preferred mapping if it actually is in the mapping
+                            if ($alias === $preferred) {
+                                $alpha2ToAlpha3[$alpha2Code] = $preferred;
+                            }
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
The nesting level of this method is pretty deep. You can flatten it by using early exits.

```php
foreach ($metadataBundle['alias']['territory'] as $alias => $data) {
    if (3 !== \strlen($alias) || 'overlong' !== $data['reason'] || ctype_digit($alias)) {
        continue;
    }

    $alpha2Code = $data['replacement'];
    if (!isset($alpha2Codes[$alpha2Code])) {
        continue;
    }
    // …
```

Reply:
Done